OCAT/client/resources/js/components/Assessments/list.js

```diff
@@ -7,161 +7,189 @@ export function AssessmentList(){
-                            col3: assessment.instrument,
-                            col4: assessment.risk_level,
-                            col5: assessment.score+'',
-                            col6: assessment.created_at,
-                        })
-                });
-                setColumnsArray(columnedData);
-            })();
-          },[]);
+        (async function() {
+            const resData = await AssessmentService.retrieveAll();
+            const columnedData = []
+            resData.data.forEach(assessment => {
+                columnedData.push(
+                    {
+                        column1: assessment.cat_name,
+                        column2: assessment.cat_date_of_birth,
+                        column3: assessment.instrument,
+                        column4: assessment.risk_level,
+                        column5: assessment.score+'',
+                        column6: assessment.created_at,
+                    })
+            });
+            setColumnsArray(columnedData);
+        })();
+    },[]);
+        
+    const tableColumns = createTableColumns();
+    const tableData = createTableDate();
+    const table = createTable(tableColumns, tableData);
+    const alternativeDisplay = createAlternativeDisplay();
+
+    let display = columnsArray.length>0 ? table: alternativeDisplay;
+    
+    return display;
+    
+    function createTableColumns() {
+        const columns = React.useMemo(
+            () => [
+                {
+                Header: 'Name',
+                accessor: 'column1',
+                sortType: 'basic',
+                Filter: columnFilter
+                },          
+                {
+                Header: 'Date of Birth',
+                accessor: 'column2',
+                sortType: 'basic',
+                Filter: columnFilter
+                }, 
+                {
+                Header: 'Instrument',
+                accessor: 'column3', 
+                sortType: 'basic',
+                Filter: columnFilter
+                },
+                {
+                Header: 'Risk Level',
+                accessor: 'column4',
+                sortType: 'basic',
+                Filter: columnFilter
+                },          
+                {
+                Header: 'Score',
+                accessor: 'column5',
+                sortType: 'basic',
+                Filter: columnFilter
+                },
+                {
+                Header: 'Created at',
+                accessor: 'column6',
+                sortType: 'basic', 
+                Filter: columnFilter
+                },
+            ],
+            []
+            );
+
+        return columns;
+    }
+
+    function createTableDate() {
+        return React.useMemo(() => columnsArray,[columnsArray]);
+    }
     
-    //for now this is an 'inline' component
-    const columnFilter = ({column}) => {
+    function createTable(columns, data){
```

Comment:
Should be its own component?

Reply:
This depends.  It's a balance between readability (often it can be bothersome to have to have 4/5 open files when it all could fit in one file).  So up to you.  This file doesn't look too large